### PR TITLE
Added new Sample (TensorFlow Multinode Training with Horovod)

### PR DESCRIPTION
**Pull request by shailensobhee**
Signed-off-by: Shailen Sobhee <shailen.sobhee@intel.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue) 

## Type of change

Please delete options that are not relevant. Add a 'X' to the one that is applicable. 

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Sample Migration (Moving sample from old repository after completing checklist established)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Command Line
- [ ] oneapi-cli
- [ ] Visual Studio
- [ ] Eclipse IDE
- [ ] VSCode

# Checklist for Moving samples:
Links and Details can be found in the samples WG Teams Files. 

- [ ] Review sample design with domain reviewers https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts 
- [ ] Implement coding guidelines and ensure code quality.
- [ ] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [ ] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com
- [ ] Adhere to readme template 
- [ ] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [ ] Enforce format via clang-format config file
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth or Joe Oster. (GitHub User: JoeOster)
- [ ] Tested using Dev Cloud when applicable
- [ ] Implement fixes for ONSAM Jiras
- [ ] If you have new dependencies/binaries, inform Samples Project Manager Swapna R Dontharaju (@srdontha) or @JoeOster

### CODE REVIEW

**Review by JoeOster — CHANGES_REQUESTED** [review]
W will need a bandit report before I can make a final approval. This scan should be run after all other changes have been made,

See Bandit GitHub for pip install and other details: https://github.com/PyCQA/bandit
To run bandit: “bandit -r path/to/your/code”

Include the report as a comment here and put on the Teams site

**Comment by shailensobhee** [comment]
Latest `bandit` report:
bandit -r Distributed_Tensorflow_Horovod/
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.6
Run started:2020-09-30 18:49:38.714428

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 129
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):

**Review by JoeOster — APPROVED** [review]

**Comment by tomlenth** [comment]
this is in pretty good shape.  a couple recommendations: 
Move the first paragraph "TensorFlow is a widely-used..." to be the first paragraph in the purpose.  it makes it easier to quickly scan.  The second paragraph "Horovod is a distributed..." could also be used in the json sample description. 

I'm a little confused by the activate conda section.  the "please replace ~/intelonepi/ for your oneapi installation path." seems like it's hanging by itself.  should it follow directly after `/intel/oneapi` in the paragraph before it?  also, oneapi should be oneAPI.

**Review by JoeOster — APPROVED** [review]
Merge approval if ci runs cleanly

**Review by JoeOster — APPROVED** [review]

**Comment by shailensobhee** [comment]
> this is in pretty good shape. a couple recommendations:
> Move the first paragraph "TensorFlow is a widely-used..." to be the first paragraph in the purpose. it makes it easier to quickly scan. The second paragraph "Horovod is a distributed..." could also be used in the json sample description.
> 
> I'm a little confused by the activate conda section. the "please replace ~/intelonepi/ for your oneapi installation path." seems like it's hanging by itself. should it follow directly after `/intel/oneapi` in the paragraph before it? also, oneapi should be oneAPI.

Implemented.

**Review by tomlenth — DISMISSED** [review]

**Review by JoeOster — APPROVED** [review]

**Comment by JoeOster** [comment]
Bandit Report uploaded to Doc Share

**Comment by JoeOster** [comment]
@anjgola - This had the Linux scan as pending since Friday I believe. Can you kick off a new validation?

**Comment by JoeOster** [comment]
@shailensobhee - Here are the error logs for linux
[19:09:22]Traceback (most recent call last):
[19:09:22]  File "oss_helper.py", line 318, in <module>
[19:09:22]    arguments.func(arguments)
[19:09:22]  File "oss_helper.py", line 265, in report_results
[19:09:22]    oss_helper.report_validation_result_for_current_os(args.commit, args.junit_xml)
[19:09:22]  File "oss_helper.py", line 155, in report_validation_result_for_current_os
[19:09:22]    pr_os_comment = commit.create_comment(os_comment_body)
[19:09:22]  File "/usr/local/lib/python3.8/dist-packages/github/Commit.py", line 164, in create_comment
[19:09:22]    headers, data = self._requester.requestJsonAndCheck(
[19:09:22]  File "/usr/local/lib/python3.8/dist-packages/github/Requester.py", line 317, in requestJsonAndCheck
[19:09:22]    return self.__check(
[19:09:22]  File "/usr/local/lib/python3.8/dist-packages/github/Requester.py", line 342, in __check
[19:09:22]    raise self.__createException(status, responseHeaders, output)
[19:09:22]github.GithubException.GithubException: 422 {"message": "Validation Failed", "errors": [{"resource": "CommitComment", "code": "custom", "field": "body", "message": "body is too long (maximum is 65536 characters)"}], "documentation_url": "https://docs.github.com/rest/reference/repos#create-a-commit-comment"}
[19:09:22]Process exited with code 1
« Hide details

**Comment by shailensobhee** [comment]
> aceback (most recent call last):
> [19:09:22] File "oss_helper.py", line 318, in
> [19:09:22] arguments.func(arguments)
> [19:09:22] File "oss_helper.py", line 265, in report_results
> [19:09:22] oss_helper.report_validation_result_for_current_os(args.commit, args.junit_xml)
> [19:09:22] File "oss_helper.py", line 155, in report_validation_result_for_current_os
> [19:09:22] pr_os_comment = commit.create_comment(os_comment_body)
> [19:09:22] File "/usr/local/lib/python3.8/dist-packages/github/Commit.py", line 164, in create_comment
> [19:09:22] headers, data = self._requester.requestJsonAndCheck(
> [19:09:22] File "/usr/local/lib/python3.8/dist-packages/github/Requester.py", line 317, in requestJsonAndCheck
> [19:09:22] return self.__check(
> [19:09:22] File "/usr/local/lib/python3.8/dist-packages/github/Requester.py", line 342, in __check
> [19:09:22] raise self.__createException(status, responseHeaders, output)
> [19:09:22]github.GithubException.GithubException: 422 {"message": "Validation Failed", "errors": [{"resource": "CommitComment", "code": "custom", "field": "body", "message": "body is too long (maximum is 65536 characters)"}], "documentation_url": "[https://docs.github.com/rest/reference/repos#create-a-commit-comment"}](https://docs.github.com/rest/reference/repos#create-a-commit-comment%22%7D)
> [19:09:22]Process exited with code 1
> « Hide details

This does not seem to be a bug with my code. I have one Python script. Is something wrong with my json file? The json validation passed OK. Can you please assist me further? The stacktrace mentions json in some places, but does not help me in getting clues on the fix.

The code runs file with oneAPI beta9 on local machines as well as on devcloud. How the CI works behind the scenes is not clear to me.

File: Frameworks/TensorFlow/Distributed_Tensorflow_Horovod/sample.json
Comment (anjgola) [inline Frameworks/TensorFlow/Distributed_Tensorflow_Horovod/sample.json:15]: here.

**Review by anjgola — APPROVED** [review]

File: Frameworks/TensorFlow/Distributed_Tensorflow_Horovod/TensorFlow_Multinode_Training_with_Horovod.py
Comment (anjgola) [inline Frameworks/TensorFlow/Distributed_Tensorflow_Horovod/TensorFlow_Multinode_Training_with_Horovod.py:2]: this file should be deleted.

**Pull request merged**
